code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -269,8 +269,11 @@
 			candidates -= player
 		else if(is_centcom_level(player.z))
 			candidates -= player
-		else if(player.mind && (player.mind.special_role || player.mind.antag_datums?.len > 0))
-			candidates -= player
+		else if(player.mind)
+			if(player.mind.assigned_role.title in restricted_roles)
+				candidates -= player
+			else if(player.mind.special_role || player.mind.antag_datums?.len > 0)
+				candidates -= player
 		else if(HAS_TRAIT(player, TRAIT_MINDSHIELD))
 			candidates -= player
```

Comment:
small bug here.

Reply:
Do tell?

Reply:
i'm a little rusty, but won't we never reach this point if the player.mind check passes?

Reply:
that would be embarrassing, you're correct